macros/edr/system/system_utils/daily_buckets_cte.sql

```diff
@@ -31,4 +31,18 @@
         from unnest(generate_timestamp_array({{ elementary.cast_as_timestamp(min_bucket_end) }}, {{ elementary.cast_as_timestamp(max_bucket_end) }}, interval 1 day)) as edr_daily_bucket
     {%- endset %}
     {{ return(daily_buckets_cte) }}
+{% endmacro %}
+
+{% macro redshift__daily_buckets_cte() %}
+    {%- set max_bucket_end = "'"~ run_started_at.strftime("%Y-%m-%d 00:00:00") ~"'" %}
+    {%- set days_back = elementary.get_config_var('days_back') %}
+
+    {%- set daily_buckets_cte %}
+        {%- for i in range(0, days_back+1) %}
+            {%- set daily_bucket = "'"~ (run_started_at - modules.datetime.timedelta(i)).strftime("%Y-%m-%d 00:00:00") ~"'" %}
+            select {{ elementary.cast_as_timestamp(daily_bucket) }} as edr_daily_bucket
```

Comment:
Try maybe using unnest and array that will be created by the macro

Reply:
Is not supported in redshift.
The union workaround is the way to do it:
https://www.holistics.io/blog/splitting-array-string-into-rows-in-amazon-redshift-or-mysql/